File: Difficulty: Easy/Two Sum in Sorted Array/two-sum-in-sorted-array.java
class Solution {
    public ArrayList<Integer> twoSum(int[] arr, int target) {
        // code here
        int n = arr.length;
        
        int low = 0;
        int high = n-1;
        
        ArrayList<Integer> list = new ArrayList<>();
        
        while(low < high) {
            int actual = arr[low] + arr[high];
            
            if(actual == target) {
                list.add(low+1);
                list.add(high+1);
                
                return list;
            }
            else if(actual > target) high-- ;
            else low++ ;
        }
        list.add(-1);
        list.add(-1);
        
        return list;
    }
}

File: Difficulty: Easy/Two Sum in Sorted Array/two-sum-in-sorted-array.cpp
class Solution {
  public:
    vector<int> twoSum(vector<int>& arr, int target) {
        // code here
        int n = arr.size();
        
        int low = 0;
        int high = n-1;
        
        vector<int> pair;
        
        while(low <= high) {
            
            int actual = arr[low] + arr[high];
            
            if( actual == target ) {
                pair.push_back(low+1);
                pair.push_back(high+1) ;
                
                return pair;
            }
            
            else if(actual > target) {
                high-- ;
            }
            else low++ ;
        }
        pair.push_back(-1);
        pair.push_back(-1);
        
        return pair;
    }
};